hatbot.gs

```diff
@@ -0,0 +1,175 @@
+
+//test of emojis in strings
+function emoji_test(){
+  var text_out = bot_name + write_tweet();
+  Logger.log(text_out);
+}
+
+//twitter api keys
+var CONSUMER_KEY = '***';
+var CONSUMER_SECRET = '***';
+
+//twitter log in
+function getService() {
+  return OAuth1.createService('Twitter')
+      // Set the endpoint URLs.
+      .setAccessTokenUrl('https://api.twitter.com/oauth/access_token')
+      .setRequestTokenUrl('https://api.twitter.com/oauth/request_token')
+      .setAuthorizationUrl('https://api.twitter.com/oauth/authorize')
+
+      // Set the consumer key and secret.
+      .setConsumerKey(CONSUMER_KEY)
+      .setConsumerSecret(CONSUMER_SECRET)
+
+      // Set the name of the callback function in the script referenced
+      // above that should be invoked to complete the OAuth flow.
+      .setCallbackFunction('authCallback')
+  
+      // Using a cache will reduce the need to read from 
+      // the property store and may increase performance.
+      .setCache(CacheService.getUserCache())
+
+      // Set the property store where authorized tokens should be persisted.
+      .setPropertyStore(PropertiesService.getUserProperties());
+}
+
+function reset() {
+  var service = getService();
+  service.reset();
+}
+
+function authCallback(request) {
+  var service = getService();
+  var authorized = service.handleCallback(request);
+  if (authorized) {
+    return HtmlService.createHtmlOutput('Success!');
+  } else {
+    return HtmlService.createHtmlOutput('Denied');
+  }
+}
+
+function get_weather_data(location) {
+  
+  var response = UrlFetchApp.fetch("api.openweathermap.org/data/2.5/weather?q="+ location +"&appid=26d507c1547e5aba63c27c987cd19aa2");
+  var json = response.getContentText();
+  var data = JSON.parse(json);
+
+  Logger.log(data);
+
+  return data;
+}
+
+function hat_or_no_hat() {
+  var cur_weather = get_weather_data(where_am_i);
+  var decision = 0;
+  //strip data
+  var tempK = cur_weather["main"]["temp"];
+  var wind = cur_weather["wind"]["speed"];
+  
+  //fix units and clean data
+  tempK = tempK.toFixed(2);
+  var tempC = (tempK - 273.15).toFixed(2);
+
+  windKM = (wind * 3.6).toFixed(0)
+
+  //make a decision using if's and comparisons
+
+  if (tempC > too_hot_for_hat) {
+    hat_hot = true;
+  }
+  else {
+    hat_hot = false;
+  }
+  if (windKM > too_windy_for_hat) {
+    hat_windy = true;
+  }
+  else {
+    hat_windy = false;
+
+  }
+  // hot and windy = 1
+  // hot and calm = 2
+  // cool and windy = 3
+  // cooooooool and caaaaaaaaalm = 4
+  if ( hat_hot == true && hat_windy == true ){decision = 1; }
+  if ( hat_hot == true && hat_windy == false ){decision = 2; }
+  if ( hat_hot == false && hat_windy == true ){decision = 3; }
+  if ( hat_hot == false && hat_windy == false ){decision = 4; }
+  
+  Logger.log(tempC);
```

Comment:
This can be consolidated into one line, possisbly making it easier to read in the logs.

It would be something like this: 
`Logger.log("temp: %d, wind: %s, too_hot: %s, too_windy: %s, descion: %s, tempC, windKM, hat_hot, hat_windy, decision);`

Where %s injects the string variable, and %d injects a number